test/test_torch.py

```diff
@@ -1909,6 +1909,26 @@ def _test_diagonal(self, dtype, device):
     def test_diagonal(self):
         self._test_diagonal(self, dtype=torch.float32, device='cpu')
 
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    def test_diagonal_multidim(self):
+        x = torch.randn(10, 11, 12, 13)
```

Comment:
Can you also add a test where x is non-contiguous, like a permutation of what you have now?